lib/pagy/frontend.rb

```diff
@@ -10,10 +10,23 @@ class Pagy
   # see https://ddnexus.github.io/pagy/api/frontend#i18n
   I18n = eval(Pagy.root.join('locales', 'utils', 'i18n.rb').read) #rubocop:disable Security/Eval
 
+  module Helpers
+    # This works with all Rack-based frameworks (Sinatra, Padrino, Rails, ...)
+    def pagy_url_for(page, pagy, path_or_url=:path)
+      p_vars = pagy.vars; params = request.GET.merge(p_vars[:page_param].to_s => page).merge!(p_vars[:params])
```

Comment:
can avoid a merge:
```suggestion
      p_vars = pagy.vars; params = request.GET.merge(p_vars[:params])
      params[p_vars[:page_param].to_s] = page
```

Reply:
we didn't notice that for so long! Thank